cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -566,3 +581,27 @@ func readCA(file string) ([]byte, error) {
 
 	return rootCA, err
 }
+
+func shouldTurnOnDynamicClient(client clientset.Interface) bool {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.TokenRequest) {
+		return false
+	}
+	apiResourceLists, err := client.Discovery().ServerResources()
```

Comment:
prefer `ServerResourcesForGroupVersion(v1.SchemeGroupVersion.String())`, which only requires a single discovery API call and doesn't require hitting any aggregated APIs

Reply:
OK